Create the k3s package using the upstream installer script

jimmykarily opened the pull request:
Hoping to replace [these commands](https://github.com/kairos-io/provider-kairos/blob/9f56a5578bfee5d0d0018d4eba7686a19939f7fb/Earthfile#L168-L172) with a simple luet install.

jimmykarily commented:
CI fails. It's reproducible locally with:

```
sudo luet build --tree packages k8s/k3s
```
not sure what's wrong yet.

Itxaka commented:
` INFO   Error response from daemon: dockerfile parse error line 15: unknown instruction: EXPORT` ???

Itxaka commented on packages/k8s/k3s/build.yaml:
```diff
@@ -0,0 +1,31 @@
+  - INSTALL_K3S_SKIP_ENABLE="true"
+  - INSTALL_K3S_SKIP_SELINUX_RPM="true"
+prelude:
+{{$arch:=(default .Values.arch "amd64")}}
```
could this be the issue? After all this is a templating instruction so the prelude will be empy

Itxaka reviewed (approved):
nice

jimmykarily commented:
The steps are run as dockerfile `RUN` entries. I moved the setting of env vars in the `env:` section and fixed that issue. Not there is another problem, that the upstream installer script cannot detect systemd/openrc inside a docker image:

```
       #9 [5/9] RUN bash installer.sh
 INFO   #9 0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s
       #9 ERROR: process "/bin/sh -c bash installer.sh" did not complete successfully: exit code: 1
       ------
        > [5/9] RUN bash installer.sh:
       0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s

```

Itxaka commented:
> The steps are run as dockerfile `RUN` entries. I moved the setting of env vars in the `env:` section and fixed that issue. Not there is another problem, that the upstream installer script cannot detect systemd/openrc inside a docker image:
> 
> ```
>        #9 [5/9] RUN bash installer.sh
>  INFO   #9 0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s
>        #9 ERROR: process "/bin/sh -c bash installer.sh" did not complete successfully: exit code: 1
>        ------
>         > [5/9] RUN bash installer.sh:
>        0.297 [ERROR]  Can not find systemd or openrc to use as a process supervisor for k3s
> ```

Im guessing you can get it to detect systemd by setting the `image` key as it was before so the base system has systemd or openrc or whatever.


Now Im guessing that is why the check for init.systemd was in there lol.

But it could also mean that we need two packages? one for systemd and one for openrc?

jimmykarily commented:
For now I opted for installing service files for both systemd (openrc and systemd). No OS would ever consume both so just having the other one laying around shouldn't be a problem (or is it?)

jimmykarily commented:
This is my ugly hack: https://github.com/kairos-io/packages/pull/335/files#diff-c9743e80f8388cd7b60efc400366bd8ef48a50f64f8c1e70bdb9c085d2f9cb4bR20-R30

Itxaka commented on packages/k8s/k3s/build.yaml:
```diff
@@ -0,0 +1,38 @@
+  - curl -sfL https://get.k3s.io > installer.sh
+  # Let the installer script install service files both for openrc and systemd:
+  # https://github.com/k3s-io/k3s/blob/36645e7311e9bdbbf2adb79ecd8bd68556bc86f6/install.sh#L114-L122
+  - touch /bin/systemctl && chmod +x /bin/systemctl
```
:rofl:

jimmykarily replied:
Poor checks need poor hacks :p

Itxaka reviewed (approved):
if it works, it works!

jimmykarily commented on packages/utils/k3d/build.yaml:
```diff
@@ -3,8 +3,7 @@ requires:
 prelude:
-{{$arch:=(default .Values.arch "amd64")}}
-
+{{$arch:=(default "amd64" .Values.arch)}}
```
This is Spring right? http://masterminds.github.io/sprig/defaults.html

Itxaka replied:
yup! https://github.com/mudler/luet/blob/master/go.mod#L6
